contrib/pkg/input/input_reconciler.go

```diff
@@ -110,31 +111,42 @@ func (r *inputReconciler) reconcileEventsForever(reconcileInterval time.Duration
 
 // processNextWorkItem deals with one key off the queue.  It returns false when it's time to quit.
 func (r *inputReconciler) processNextWorkItem() bool {
-	key, quit := r.queue.Get()
+	queueItem, quit := r.queue.Get()
 	if quit {
 		return false
 	}
-	defer r.queue.Done(key)
+	defer r.queue.Done(queueItem)
 
-	var isRemoteCluster bool
+	// get the string representation of the queue item
+	key, ok := queueItem.(string)
+	if !ok {
+		contextutils.LoggerFrom(r.ctx).Errorw("got a work queue item of non-string type", "item", queueItem)
+		r.queue.Forget(queueItem)
```

Comment:
should we return after the forgets?

Reply:
new line 159 is showing that this is the default behavior so I'd think we would want to

Reply:
ha yeah good catch, both places where we error, we should return